engine/jobmaster/dm/dm_jobmaster.go

```diff
@@ -128,7 +131,11 @@ func (jm *JobMaster) InitImpl(ctx context.Context) error {
 	if err := jm.checkpointAgent.Create(ctx, jm.initJobCfg); err != nil {
 		return err
 	}
-	return jm.taskManager.OperateTask(ctx, dmpkg.Create, jm.initJobCfg, nil)
+	if err := jm.taskManager.OperateTask(ctx, dmpkg.Create, jm.initJobCfg, nil); err != nil {
+		return err
+	}
+	jm.initialized.Store(true)
```

Comment:
`OnMasterRecovered` should also set `initialized`, because either `InitImpl` or `OnMasterRecovered` is called during job master starts up

https://github.com/pingcap/tiflow/blob/04173f64ea66bd4e709f8274e5bd1ffe60e70625/engine/framework/base_jobmaster.go#L202-L217

Reply:
Is it possible that `InitImpl` and `OnMasterRecovered` are called on the same `JobMasterImpl` struct?

Reply:
No, they are mutually exclusive.

Reply:
Addressed in https://github.com/pingcap/tiflow/pull/7587/commits/fee52ba544791a0b863d51137cc5d20c82e47528.